### PR TITLE
Code Review 2 changes

### PR DESCRIPTION
**Pull request by JoshuaGyau**
- Analysis of the program.
Their project is looking good so far. their UI just needs to be organized and little better. the placement of the icons, the spacing and little things like that.  but overall, the project is coming together very nicely 

- Was the program available in UC Github on time?
Yes it was available on time

- Is the program easy for you, as an outsider, to understand?
yes, they left a lot of comments, used good naming conventions. the readme was clear, the requirements made sense. the whole projects was simple to follow. 

- Does the program compile?
yes the program compiles 

- Rationale Behind Changes
I removed unused imports
I removed commented code that is no longer necessary
changed some of the variables to camel case
changed the visibility modifiers for a some variables and methods.

- What I learnt 

https://kotlinlang.org/docs/visibility-modifiers.html#packages
I learned about the internal visibility modifier. I saw it used for the first time in their code, even though it was not needed in that instance. 

they have a search Icon that displays the textbox after it is clicked. I like that idea a lot, and will be adding something similar to my project. 

I also learned how to change the fragment by looking at how they change to the details fragment whenever a recipe is selected. this is something  I have recently been working on for my project so it was very helpful.

### CODE REVIEW

File: app/src/main/java/com/main/mealplanner/MainActivity.kt [hunk 1]
```diff
@@ -2,12 +2,8 @@ package com.main.mealplanner
 
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
-import android.widget.AutoCompleteTextView
-import android.widget.SearchView
```
Comment (JoshuaGyau) [inline app/src/main/java/com/main/mealplanner/MainActivity.kt:6]: removed unused imports

File: app/src/main/java/com/main/mealplanner/MainViewModel.kt [hunk 2]
```diff
@@ -32,15 +29,5 @@ class MainViewModel : ViewModel() {
...
-//         recipe.recipeID = document.id
-//         val set = document.set(recipe)
-//            set.addOnSuccessListener { it: Void!
-//                Log.d(tag: "Firebase", msg: "document saved")
-//            }
-//            set.addOnFailureListener { it: Exception
-//                Log.d(tag: "Firebase", msg: "save failed")
-//                                     }
-//     }
+
```
Comment (JoshuaGyau) [inline app/src/main/java/com/main/mealplanner/MainViewModel.kt:32]: remove unnecessary commented code

File: app/src/main/java/com/main/mealplanner/dao/IMealPlanDAO.kt [hunk 3]
```diff
@@ -11,11 +11,11 @@ interface IMealPlanDAO {
     fun getAllMealPlans()  : LiveData<List<MealPlan>>
 
     @Insert(onConflict = OnConflictStrategy.REPLACE)
-    fun insertAll(mealplans: ArrayList<MealPlan>)
+    fun insertAll(mealPlan: ArrayList<MealPlan>)
 
     @Delete
-    fun delete(mealplan: MealPlan)
+    fun delete(mealPlan: MealPlan)
```
Comment (JoshuaGyau) [inline app/src/main/java/com/main/mealplanner/dao/IMealPlanDAO.kt:17]: change to camel case

File: app/src/main/java/com/main/mealplanner/RetrofitClientInstance.kt [hunk 4]
```diff
@@ -5,7 +5,7 @@ import retrofit2.converter.gson.GsonConverterFactory
 
 object RetrofitClientInstance {
     private var retrofit: Retrofit? = null;
-    private val BASE_URL = "https://www.themealdb.com/api/json/v1/1/"
+    private const val BASE_URL = "https://www.themealdb.com/api/json/v1/1/"
```
Comment (JoshuaGyau) [inline app/src/main/java/com/main/mealplanner/RetrofitClientInstance.kt:8]: change url variable to a constant

File: app/src/main/java/com/main/mealplanner/UI/DetailsFragment.kt [hunk 5]
```diff
@@ -1,28 +1,22 @@
...
 import com.main.mealplanner.dto.RecipeDetails
 import com.squareup.picasso.Picasso
 import kotlinx.android.synthetic.main.details_fragment.*
-import kotlinx.android.synthetic.main.main_fragment.*
-import java.util.ArrayList
+
 
 open class DetailsFragment: Fragment(){
-    internal lateinit var viewModel: MainViewModel
+    private lateinit var viewModel: MainViewModel
```
Comment (JoshuaGyau) [inline app/src/main/java/com/main/mealplanner/UI/DetailsFragment.kt:19]: Changed internal to private

File: app/src/main/java/com/main/mealplanner/MainViewModel.kt (hunk 2 above)
Comment (discospiff) [reply app/src/main/java/com/main/mealplanner/MainViewModel.kt:32]: Good.

File: app/src/main/java/com/main/mealplanner/dao/IMealPlanDAO.kt (hunk 3 above)
Comment (discospiff) [reply app/src/main/java/com/main/mealplanner/dao/IMealPlanDAO.kt:17]: Good.

**Comment by discospiff** [comment]
These changes reduce technical debt.   It is worth merging, or implementing specific changes manually.  (Organize imports can be done automatically by IDE.)

**Comment by Brad-Davidson** [comment]
I'm merging this branch because it seems to reduce the amount of unneeded importants. The code for the "save" function was commented out for testing stuff, but it seems as we may be going in a new direction anyways. Thanks for the code review!

**Pull request merged**